Feat: SSE 알림 기능 추가 BACK-214

ah9mon opened the pull request:
- SSE 알림객체인 Notification 객체 추가
- Notification 관련 Dto Repository 등 추가
- 아이템 출시시 알림 기능 추가
- Profile 생성시 0원인 아이템 모두 추가
- 0원인 아이템 중 hamster를 기본 프로필로 지정
- 추후 제공하는 기본 프로필 변경, 추가를 생각해 enum 객체 BasicProfileItem 추가

ah9mon committed:
Feat: SSE 알림 기능 추가

- SSE 알림객체인 Notification 객체 추가
- Notification 관련 Dto Repository 등 추가
- 아이템 출시시 알림 기능 추가

ah9mon committed:
Fix: 알림서비스 코드 수정

ah9mon committed:
Fix: 읽음여부 추가 BACK-214

Fishphobiagg commented on src/main/java/com/siliconvalley/domain/notification/dao/NotificationFindDao.java:
아직 작성이 안된 클래스는 제외시키는게 좋을 것 같습니다!

Fishphobiagg commented on src/main/java/com/siliconvalley/domain/notification/domain/NotificationType.java:
```diff
@@ -0,0 +1,19 @@
+package com.siliconvalley.domain.notification.domain;
+
+public enum NotificationType {
```
알림 템플릿을 관리하는 방식을 잘 정하신 것 같습니다! 프로젝트의 요구사항에 맞게 enum으로 관리하는 것이 적절해 보입니다!

Fishphobiagg commented on src/main/java/com/siliconvalley/domain/profile/domain/BasicProfileItem.java:
```diff
@@ -0,0 +1,15 @@
+package com.siliconvalley.domain.profile.domain;
+
+public enum BasicProfileItem {
```
enum을 통해 객체의 책임과 행동에 대한 추가, 또 변경이 잦지 않을 것 같은 데이터를 enum으로 관리하는 방식이 매우 좋은 것 같습니다! 
저도 영감을 얻고 좋은 글을 하나 찾아봤는데, enum을 적용할때 아래 글 참고해보시면 좋을 것 같습니다!

https://techblog.woowahan.com/2527/

ah9mon committed:
Feat: Notification 관련 API 및 랭킹 알림 추가

- Notification 상세조회
- Notification 전체 조회 페이지네이션
- Notification 삭제
- 랭킹에 등록된 사용자에게 알림 보내기

Fishphobiagg reviewed (approved):
GOOD!!! 코드가 가독성 좋게 잘 작성되어있는 것 같습니다
